### DISCUSSION TITLE
How to migrate standalone k3s to existing External datastore backed K3s Cluster (multi-node)

### DISCUSSION THREAD

**Discussion by prnvkv** [k3s-io/k3s#4650] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
This is not a feature request but just a question. If we have a k3s single node cluster running on a local datastore (default), how can we migrate to existing k3s multi-node cluster running with external datastore (e.g. MariaDB) without affecting the running workloads.

**Describe the solution you'd like**
When a k3s standalone node joins the existing cluster, the existing workloads should not be affected. 

**Describe alternatives you've considered**
No alternative.

**Comment by brandond** [comment, marked as answer]
We don't currently support migrating from one Kine sql backend to another, say from sqlite to mysql. We do support migrating from sqlite to etcd; all you need to do is start an existing standalone node with --cluster-init.

**Comment by ysicing** [comment]
Is it possible to manually migrate data from SQLite to MySQL? @brandond

**Reply by brandond** [reply]
Anything is possible. Do I have a recommendation for a tool to migrate tables from sqlite to mysql? No.

**Reply by ysicing** [reply]
> Anything is possible. Do I have a recommendation for a tool to migrate tables from sqlite to mysql? No.

Haha? Of course, PG tools are also fine, I'm very happy to try.
